src/components/Form/Form.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import formStyles from "./form.module.css";
+import cartStyles from "../Cart/cart.module.css";
+import { db } from "../../firebase/config";
+
+const Form = ({ setSubmitted, cartItems }) => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [adress, setAdress] = useState("");
+  const [city, setCity] = useState("");
+  const [country, setCountry] = useState("");
+  const [zip, setZip] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone, setPhone] = useState("");
+
+  const order = cartItems.map((item) => `${item.name}`).toString();
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    setSubmitted(true);
```

Comment:
setSubmitted() should be called _after_ the firebase request. Otherwise, you are showing the success message without the data being really sent to the server.

So, you should move it to the callback next to the successfully log.

---

Note: I believe you can use firebase with async await too!


```js
await  function handleSubmit(e) {
try {
    await db.ref("orders/" + Date.now()).set({
      ...formValues,
    })

    setSubmitted(true);
} catch (exception) {
  // show the error message
} 
}
```

Reply:
Makes perfect sense. Thanks again! 😄